### PR TITLE
[4.0] [com_csp] Show a warning in case we are collecting but there are trashed itms

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #25594 cc @brianteeman 

### Summary of Changes

show a warning in case we are collecting but there are trashed itms

### Testing Instructions

- Enable the collecting mode of com_csp
- collect some reports
- trash a few or all of them
- refresh the reports list

### Expected result

You should now get a warning indicating that we are collecting but have also trashed items

### Actual result

No indication that we have trashed items.

### Documentation Changes Required

none.

### Additional Info

@brianteeman  I'm not happy with the warning message yet, you might can suggest a bettwer wording here? Thanks.

### CODE REVIEW

File: administrator/components/com_csp/Helper/ReporterHelper.php [hunk 1]
```diff
@@ -48,4 +48,32 @@ public static function getHttpHeadersPluginId()
 
 		return $result;
 	}
+
+	/**
+	 * Check the com_csp trash to show a warning in this case
+	 *
+	 * @return  boolean  The status of the trash; Does items exists in the trash
```
Comment (brianteeman) [inline administrator/components/com_csp/Helper/ReporterHelper.php]:
```suggestion
	 * @return  boolean  The status of the trash; Do items exists in the trash
```

File: administrator/language/en-GB/en-GB.com_csp.ini [hunk 2]
```diff
@@ -4,6 +4,7 @@
 ; Note : All ini files need to be saved as UTF-8
 
 COM_CSP="Content Security Policy"
+COM_CSP_COLLECTING_TRASH_WARNING="You are running your rules in the collect mode. Please note that there are trashed items that don't get colleted here again untill fully deleted."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_csp.ini]: its "detect" not "collect" mode ;)

```suggestion
COM_CSP_COLLECTING_TRASH_WARNING="The Content Security Policy is in detect mode. Items that have been trashed will not be detected again until they are removed from the trash."
```

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 024acdacd14a70deb5a135ea32050c21fda5a2e6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25712">issues.joomla.org/tracker/joomla-cms/25712</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 024acdacd14a70deb5a135ea32050c21fda5a2e6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25712">issues.joomla.org/tracker/joomla-cms/25712</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**